pkg/platform/runtime/runtime.go

```diff
@@ -63,11 +75,33 @@ func newRuntime(target setup.Targeter, an analytics.Dispatcher, svcModel *model.
 		}
 	}
 
+	if target.ProjectDir() == "" || auth == nil {
+		return rt, nil
+	}
+
+	// Check if local build script has changes that should be staged.
+	if script, err := buildscript.NewScriptFromProjectDir(target.ProjectDir()); err == nil {
+		bp := model.NewBuildPlannerModel(auth)
+		if commitID, err := localcommit.Get(target.ProjectDir()); err == nil {
+			if expr, err := bp.GetBuildExpression(target.Owner(), target.Name(), commitID.String()); err == nil {
+				if !script.Equals(expr) {
+					return rt, &NeedsStageError{errs.New("Runtime changes should be staged")}
+				}
+			} else {
+				multilog.Error("Unable to get remote build expression: %v", errs.JoinMessage(err))
+			}
+		} else if !localcommit.IsFileDoesNotExistError(err) {
```

Comment:
This won't ever trigger cause line 85 is checking for non-nil errors, meaning this else would only get nil errors.

Reply:
Looks okay to me. Must have been a trick of the light. Line 85 is checking for `err == nil`.

Reply:
My bad, I did indeed read that as `!= nil`.